controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -62,7 +77,11 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 
-	if len(storageClusterList.Items) > 1 {
+	switch l := len(storageClusterList.Items); {
+	case l == 0:
+		duration := time.Minute
+		return reconcile.Result{RequeueAfter: duration}, fmt.Errorf("no StorageCluster found, waiting for %d to be created", duration)
+	case l != 1:
 		return reconcile.Result{}, fmt.Errorf("multiple StorageCluster found")
 	}
```

Comment:
This will not work as you expect. If you return an error the result will be ignored (with its configuration), so `duration` will be ignored. 

Also' we will need the storage cluster object throughout the reconciliation, so let's save it on the reconciler instead of passing it via args

```suggestion
	scCount := len(storageClusterList.Items);
	if scCount == 1 {
		r.storageCluster = &storageClusterList.Items[0]
	} else if scCount == 0 {
		return reconcile.Result{}, fmt.Errorf("Storage cluster not found")
	} else {
		return reconcile.Result{}, fmt.Errorf("multiple StorageCluster found")
	}
```

As a side note, I also think that we should also put the context on the reconciler, again this way we will not need to pass it through arguments. So at the start of the function we will do:
```
	r.ctx = ctx
```